app/scripts/controllers/containerfileviewer.js

```diff
@@ -66,6 +85,81 @@ angular.module('dockstore.ui')
           function(result){
             $scope.selTagName = $scope.successContent[0].tag;
             $scope.selDescriptorName = $scope.successContent[0].descriptor;
+            $scope.fileContent = $scope.successContent[0].content;
+            var result = $scope.fileContent.toLowerCase();
+            m = [];
+            v = false;
+            if($scope.selDescriptorName === "cwl"){
+              //Descriptor: CWL
+              if(result.search("inputs:") !== -1){
```

Comment:
I'm afraid moving it to a function and calling it from there wont really work. If you see there, the validation check is inside a returned promise and this process i asynchronous. So, moving it to a function and calling it from there wont work in the end because it will be async. However, I'm going to try to move the required fields into an array and check it through a for loop, so the code will be way shorter in the end.